plugins/state_history_plugin/tests/CMakeLists.txt

```diff
@@ -1,6 +0,0 @@
-add_executable( test_state_history main.cpp session_test.cpp plugin_config_test.cpp)
-target_link_libraries(test_state_history state_history_plugin eosio_testing eosio_chain_wrap)
-target_include_directories( test_state_history PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/../include" )
-
-add_test(test_state_history test_state_history)
-set_property(TEST test_state_history PROPERTY LABELS nonparallelizable_tests)
```

Comment:
I feel bad for the removal of these tests. Unfortunately many of them, due to how the mocking is structured, would need to be significantly rewritten. They also impose an additional 'non-natural' burden on the code, such as needing to support uncompressed logs -- something that _never_ actually happens in practice.

I've added #230 to contemplate what to do about restoring these.

imo the biggest loss is likely any tests around nuanced semantics of how requests are handled. For example, how `have_positions` operates. Otherwise, our integration tests bang on most aspects of ship's connection.

I took the current `have_positions` logic from 3.x which might not be exactly how 4 or 5 performs it, I'm unsure. It's a little difficult to ascertain what clients expect with `have_positions` because I'm not aware of any client that uses that functionality.